lms/djangoapps/courseware/views.py

```diff
@@ -1311,3 +1314,31 @@ def generate_user_cert(request, course_id):
     # at backend debug the issue and re-submit the task.
 
     return HttpResponseBadRequest(_("Creating certificate"))
+
+
+def _track_successfull_certificate_generation(user_id, course_id):
+    """Track an successfully certificate generation event.
+
+    Arguments:
+        user_id (str): The ID of the user generting the certificate.
+        course_id (unicode):  id associated with the course
+    Returns:
+        None
+
+    """
```

Comment:
Please add this line to avoid firing events if Segment isn't configured:
```
if settings.FEATURES.get('SEGMENT_IO_LMS') and hasattr(settings, 'SEGMENT_IO_LMS_KEY'):
    return
```